wordpress-mobile/WordPress-iOS
# Gutenberg integration

This PR merges Gutenberg into develop.

- The display of Gutenberg as the posts/pages editor is behind a feature flag and a launch argument. (#10497)
- The only action required for anybody to continue working normally is to run `pod install`.
- If anyone wants to try Gutenberg, is necessary to switch the launch argument ON (the default is off).
- This launch argument will be in place until we implement the in-app switch.

There are still some small details that we will be working on in the remaining time.
The most remarkable is that the `Before you continue...` alert is still shown when the user opens a Gutenberg post with Gutenberg.

**Note:**
There is a bug on React Native that when running on iOS 12, there's a continuous stream of logs referring to `nw_socket_handle_socket_event [] Socket SO_ERROR`. **This will happen only if and after you open the Gutenberg editor**.

This is a known issue by the RN team and I hope they will fix it soon.
(https://github.com/facebook/react-native/issues/21030)

### To test:

@loremattei Please test the release flow to make sure that everything continues working for normal releases.
It would be great to make a Gutenberg Alpha release test running Gutenberg to see if that crash you mentioned before continues happening. If that is the case, a crash log would be super helpful.
- As mentioned by Koke here https://github.com/wordpress-mobile/WordPress-iOS/pull/10520#issuecomment-441011076, you might want to modify the feature flag to test the Alpha release.

@diegoreymendez If you could check the changes made to `AztecPostViewController` would be great. It wasn't more than moving code to share it with `GutenbergViewController`.
If you don't feel confortable with those changes, we still can reset `AztecPostViewController` back.

cc @koke @hypest @pinarol
